dockerfiles/Dockerfile.training

```diff
@@ -0,0 +1,197 @@
+ARG MOFED_VERSION=4.7-3.2.9.0
+ARG UCX_VERSION=1.8.0
+ARG OPENMPI_VERSION=4.0.4
+ARG CONDA_VERSION=4.7.10
+ARG MKL_VERSION=2020.1
+ARG NUMPY_VERSION=1.18.5
+ARG ONNX_VERSION=1.7.0
+
+# cuda development image for building sources
+FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04 as build
+
+# set location for builds
+WORKDIR /stage
+
+# install curl
+RUN apt-get -y update &&\
+    apt-get -y --no-install-recommends install \
+        curl
+    
+# install miniconda (comes with python 3.7 default)
+ARG CONDA_VERSION
+ARG CONDA_URL=https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-x86_64.sh
+RUN cd /stage && curl -fSsL --insecure ${CONDA_URL} -o install-conda.sh &&\
+    /bin/bash ./install-conda.sh -b -p /opt/conda &&\
+    /opt/conda/bin/conda clean -ya
+ENV PATH=/opt/conda/bin:${PATH}
+
+# install mkl, numpy, and pytorch build dependencies
+ARG MKL_VERSION
+ARG NUMPY_VERSION
+ARG ONNX_VERSION
+RUN conda install -y \
+        mkl=${MKL_VERSION} \
+        mkl-include=${MKL_VERSION} \
+        numpy=${NUMPY_VERSION} \
+        cmake \
+        ninja \
+        pyyaml \
+        cffi \
+        setuptools &&\
+    pip install \
+        onnx=="${ONNX_VERSION}"
+
+# install mellanox libraries for ib etc. (see below notes for user requirements)
+ARG MOFED_VERSION
+ARG MOFED_OS=ubuntu18.04
+ARG MOFED_TARNAME=MLNX_OFED_LINUX-${MOFED_VERSION}-${MOFED_OS}-x86_64
+ARG MOFED_URL=https://www.mellanox.com/downloads/ofed/MLNX_OFED-${MOFED_VERSION}/${MOFED_TARNAME}.tgz
+RUN cd /stage && curl -fSsL ${MOFED_URL} | tar -zxpf - &&\
+    mv ${MOFED_TARNAME} mofed-${MOFED_VERSION} &&\
+    cd mofed-${MOFED_VERSION} && \
+    ./mlnxofedinstall --basic --force --user-space-only --without-fw-update
+
+# build ucx suite (leverages mellanox libraries)
+# note: openmpi will not select ucx without multithreading enabled
+ARG UCX_VERSION
+ARG UCX_TARNAME=ucx-$UCX_VERSION
+ARG UCX_URL=https://github.com/openucx/ucx/releases/download/v${UCX_VERSION}/${UCX_TARNAME}.tar.gz
+RUN apt-get -y update && apt-get -y --no-install-recommends install \
+        libnuma-dev &&\
+    cd /stage && curl -fSsL ${UCX_URL} | tar xzf - &&\
+    cd ${UCX_TARNAME} &&\
+    ./configure \
+	--prefix=/opt/ucx \
+        --with-cuda=/usr/local/cuda \
+        --with-verbs=/usr \
+        --enable-mt &&\
+    make -j"$(nproc)" &&\
+    make install
+
+# build openmpi (use --prefix /opt/ompi4 to move to runtime image)
+# note: require --enable-orterun-prefix-by-default for Azure machine learning compute
+# note: disable verbs as we use ucx middleware and don't want btl openib warnings
+ARG OPENMPI_VERSION
+ARG OPENMPI_TARNAME=openmpi-${OPENMPI_VERSION}
+ARG OPENMPI_URL=https://download.open-mpi.org/release/open-mpi/v%OMPI_BASE%/${OPENMPI_TARNAME}.tar.gz
+RUN export OMPI_BASE=${OPENMPI_VERSION%.*} &&\
+    cd /stage && curl -fSsL `echo ${OPENMPI_URL} | sed s/%OMPI_BASE%/$OMPI_BASE/` | tar xzf - &&\
+    cd ${OPENMPI_TARNAME} &&\
+    ./configure \
+        --prefix=/opt/ompi4 \
```

Comment:
how about change to --prefix=/opt/openmpi-${OPENMPI_VERSION}?

Reply:
Yeah, can do.